Code/Framework/AzCore/AzCore/PlatformId/PlatformDefaults.cpp

```diff
@@ -15,18 +15,24 @@ namespace AZ
 {
     inline namespace PlatformDefaults
     {
-        static const char* PlatformNames[PlatformId::NumPlatformIds] = { PlatformPC, PlatformAndroid, PlatformIOS, PlatformMac, PlatformProvo, PlatformSalem, PlatformJasper, PlatformServer, PlatformAll, PlatformAllClient };
+        static const char* PlatformNames[PlatformId::NumPlatformIds] = { PlatformPC, PlatformLinux, PlatformAndroid, PlatformIOS, PlatformMac, PlatformProvo, PlatformSalem, PlatformJasper, PlatformServer, PlatformAll, PlatformAllClient };
 
         const char* PlatformIdToPalFolder(AZ::PlatformId platform)
         {
 #ifdef IOS
 #define AZ_REDEFINE_IOS_AT_END IOS
 #undef IOS
+#endif
+#ifdef LINUX
+#define AZ_REDEFINE_LINUX_AT_END LINUX
+#undef LINUX
```

Comment:
This could just use the #pragma push macro .. #undef LINUX .. #pragma pop macro logic as below
It saves a couple of lines.

Plus we can undefine it for the entire file as well
I suggest we do the same for IOS as well.
We just undefined it at the top of the file

```
#pragma push_macro("LINUX")
#undef LINUX
...
#pragma pop_macro("LINUX")
```

Reply:
I've renamed this anyway to `LINUX_ID`, so these push/pop pairs aren't necessary.